### PR TITLE
Use the context of element for events dispatching

### PR DESCRIPTION
**Pull request by stockiNail**
This PR changes the context instance to pass to event hook when the event is dispatched.

Instead of creating new object every time, the `element.$context` is used.

This enables to the users to store custom properties and to retrieve them in other event hooks.

### TODO

- [x] test case, waiting for #520 approval and merge

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -95,22 +95,22 @@ function handleClickEvents(chart, state, event, options) {
...
-      dispatchEvent(chart, click, element, event);
+      dispatchEvent(click, element, event);
     }
   }
 }
 
-function dispatchEvent(chart, handler, element, event) {
-  callHandler(handler, [{chart, element}, event]);
+function dispatchEvent(handler, element, event) {
+  callHandler(handler, [element.$context, event]);
```
Comment (kurkle) [inline src/events.js:113]: Does the element **always** contain `$context`?
I'd think it does not when there is nothing scriptable, but that maybe not the case?

Comment (kurkle) [reply src/events.js:113]: Answering to myself, its always there. And it set there (in `updateElements`):

https://github.com/chartjs/chartjs-plugin-annotation/blob/405433949306d949e4460e8e65487b4baa650346/src/annotation.js#L142

**Review by kurkle — APPROVED** [review]

File: test/events.js [hunk 2]
```diff
@@ -1,4 +1,9 @@
+const annotationTypes = ['box', 'ellipse', 'label', 'line', 'point', 'polygon'];
+
 export function testEvents(options, innerElement) {
+  if (!annotationTypes.includes(options.type)) {
+    console.warn(`No annotation found with type '${options.type}'.`);
```
Comment (kurkle) [inline test/events.js:5]: I was thinking the warning should be output when creating the annotation elements. I'll create a pr for that.

Comment (kurkle) [reply test/events.js:5]: Created #530

Comment (stockiNail) [reply test/events.js:5]: Yes, I did before that PR. I'm going to remove it

Comment (stockiNail) [reply test/events.js:5]: And also merge because a rebase is needed

Comment (stockiNail) [reply test/events.js]: fixed and rebased

**Review by kurkle — APPROVED** [review]

**Pull request merged**